tapdb/sqlc/queries/assets.sql

```diff
@@ -321,7 +321,18 @@ JOIN genesis_info_view
 -- around that needs to be used with this query until a sqlc bug is fixed.
 LEFT JOIN key_group_info_view
     ON assets.genesis_id = key_group_info_view.gen_asset_id
-WHERE spent = FALSE
+JOIN managed_utxos utxos
+    ON (assets.anchor_utxo_id = utxos.utxo_id OR assets.anchor_utxo_id IS NULL) AND
```

Comment:
The issue is that when anchoring pending assets with `AnchorPendingAssets` we select the assets on `prev_out`, but the genesis asset has a different `prev_out` than the other assets in the batch. So all assets _except_ the genesis asset is anchored.

Reply:
So `addRandGroupToBatch` creates a random asset genesis and stores it with `storeGroupGenesis`. But that random asset doesn't have a managed utxo in the database. It's inserted with `upsertAssetsWithGenesis` and although that func does allow for a `utxo_id` being passed, in this case `nil` is passed.

I have changed it, so that it _also_ creates a managed utxo in the database, and then passes the `utxo_id` it gets back from that operation to `upsertAssetsWithGenesis`. It's quite involved.

Because that random asset is now a proper asset that returns in `FetchAllAssets` it does change some of the assumptions in the original failing test `TestCommitBatchChainActions`.

What this all means is that we can have proper JOINS without the trickery from before.